roxctl/central/debug/download_diagnostics.go

```diff
@@ -58,3 +67,10 @@ func downloadDiagnosticsCommand(cliEnvironment environment.Environment) *cobra.C
 
 	return c
 }
+
+func isTimeoutError(err error) bool {
+	if err == nil {
+		return false
+	}
+	return strings.Contains(err.Error(), "net/http: request canceled") || errors.Is(err, context.DeadlineExceeded)
```

Comment:
Instead of the string checking, you can do the following (which would mean skipping the string comparison):
```
var urlError *url.Error
return errors.As(err, &urlError) && urlError.Timeout()
```

Reply:
Does not work

```
INFO:   Downloading diagnostic bundle...
ERROR:  error writing to ZIP file: net/http: request canceled (Client.Timeout or context cancellation while reading body)

```

Reply:
Ah, I initially looked at non-http2 code.

The error returned will be `x/net/http2/httpError`, which satisfies the `net.Error` interface, so this should now do:
```
var netErr net.Error
return errors.As(err, &netErr) && netErr.Timeout()
```

Reply:
@dhaus67 nice, it works :)